### ISSUE TITLE
pagevar fails to retrive variables randomly

### ISSUE THREAD

**Issue by newptcai** [tlienart/Franklin.jl#937]
I tried to update my [website](https://gitlab.com/newptcai/newptcai.gitlab.io) to use latest Franklin. But I found a weird bug. The following code failed to retrieve page variables currently.

~~~
posts = filter!(p -> endswith(p, ".md") && p != "index.md", readdir("blog"))
dates  = Vector{Date}(undef, length(posts))
lines = Vector{String}(undef, length(posts))
indices  = Vector{Tuple{Date, String}}(undef, length(posts))
for (i, post) in enumerate(posts)
    ps  = splitext(post)[1]
    url = "/blog/$ps/"
    surl = strip(url, '/')
    title = pagevar(surl, :title)
    pubdate = pagevar(surl, :pubdate)
    if isnothing(pubdate)
        dates[i] = Date("2021-01-01")
    else
        dates[i] = Date(pubdate)
    end
    lines[i] = "\n* [$title]($url)\n"
    indices[i] = (dates[i], title)
end
~~~

For some pages, `title = pagevar(surl, :title)` will return `nothing`. And for which pages this happens is random if you run `serve(clear=true)`. The line `pubdate = pagevar(surl, :pubdate)` also fails at random even when `pubdate` is defined.

To reproduce this error, you can clone my website, and update to v0.10.65. The run the website.

**Comment by tlienart** [comment]
Please use `@delay` in front of the `hfun` that makes use of all the `pagevar` so

```
@delay function hfun_***...
...
end
```

**Comment by Wikunia** [comment]
Just encountered this as well. I thought `pagevar` checks if the variable is already there and if not parses the markdown file to retrieve it. Also ` @delay` doesn't seem to work in my case.

**Comment by newptcai** [comment]
Adding @delay does not change anything for me neither.

**Comment by tlienart** [comment]
@newptcai same suggestion as I made to @Wikunia ; for now please pin Franklin to `0.10.58` if things used to work for you but don't anymore now.

The next version will be released soon and fixes this pagevar stuff so this workaround should hopefully be ok in the meantime

**Comment by newptcai** [comment]
@tlienart Any update?

**Comment by tlienart** [comment]
did you do the pinning suggested above?

**Comment by newptcai** [comment]
Yes, I pinned the version and it works now. How can I test the upcoming version?

**Comment by tlienart** [comment]
:+1: I'll close this issue.

If you want to test the new version, have a look at [this repository](https://github.com/tlienart/Xranklin.jl), please read the README as it contains information about the migration and what users should be careful about. Also if you intend to use it, please join the slack channel `#xranklin` for questions.

**Comment by tlienart** [comment]
Hello @maleadt, sorry for the frustration, I'd recommend you pin Franklin to 10.58 and check if that helps for the time being.

This problem has been fixed by overhauling the entirety of the pagevar logic in the upcoming new release which is in prep at Xranklin.jl. It's pretty much ready bar a few minor stuff I need to find some time to complete.
